libstorage/drivers/storage/cinder/storage/cinder_storage.go

```diff
@@ -672,7 +675,26 @@ func (d *driver) VolumeAttach(
 			fields, "error waiting for volume to attach", err)
 	}
 
-	return volume, volumeAttach.Device, nil
+	return volume, resolveDeviceName(volumeID, volumeAttach.Device), nil
+}
+
+func getDeviceLink(
```

Comment:
Resolving devices that are on the node by peaking in `/dev` cannot be done in this code, as it may not be running on the same node. REX-Ray supports a client/server mode of operation, where a single REX-Ray instance running in server mode is configured with the Cinder API credentials, but all client nodes are just talking to the master REX-Ray instance.

Any code that is specific to a single node/instance needs to be in the executor, not here.

The way this should be handled is in the `LocalDevices` function in the [executor](https://github.com/rexray/rexray/blob/d14c3e3073be0b0955648b7de8a0a882a87977d7/libstorage/drivers/storage/cinder/executor/cinder_executor.go#L203-L224). This function returns a map of attached devices, with their actual path to their volume ID. Right now, this function is returning all attached devices, and an empty volume ID. This would need to modified to extract the volume ID from  your `virtio` path used below...